### PR TITLE
Add header support for Domain socket

### PR DESCRIPTION
**Pull request by ccaapton**
This PR address the issue that current code base does not support headers on unix sockets.

### CODE REVIEW

**Comment by ccaapton** [comment]
The affected codacy check is due to protobuf generated code, and it should not have a real damage.

File: infra/conf/transport_internet.go [hunk 1]
```diff
@@ -210,13 +210,27 @@ func (c *QUICConfig) Build() (proto.Message, error) {
 type DomainSocketConfig struct {
 	Path     string `json:"path"`
 	Abstract bool   `json:"abstract"`
+	HeaderConfig json.RawMessage `json:"header"`
```
Comment (agioi) [inline infra/conf/transport_internet.go:213]: you should run `gofmt`

File: transport/internet/domainsocket/listener.go [hunk 2]
```diff
@@ -24,6 +24,7 @@ type Listener struct {
 	config    *Config
 	addConn   internet.ConnHandler
 	locker    *fileLocker
+	authConfig internet.ConnectionAuthenticator
```
Comment (agioi) [inline transport/internet/domainsocket/listener.go:27]: `gofmt` again

**Comment by rprx** [comment]
代码含义无问题，请格式化 + 重新生成 .pb.go 后提交至 https://github.com/v2fly/v2ray-core/pulls

**Comment by rprx** [comment]
@kslr 此 PR 可以关闭，ds 写到 listen 后 inbound 将支持 header，而 ds outbound 支持 header 似乎没有意义。

https://github.com/v2fly/v2ray-core/pull/367

**Pull request closed without merging**